plutus-tx/src/Language/PlutusTx/Applicative.hs

```diff
@@ -29,7 +32,15 @@ a1 *> a2 = (id <$ a1) <*> a2
 (<*) :: Applicative f => f a -> f b -> f a
 (<*) = liftA2 const
 
+{-# INLINABLE sequence #-}
+-- | Sequence a list of applicative actions.
+sequence :: Applicative f => [f a] -> f [a]
+sequence []    = pure []
+sequence (h:t) = (:) <$> h <*> sequence t
+
 instance Applicative Maybe where
+    {-# INLINABLE pure #-}
     pure = Just
+    {-# INLINABLE (<*>) #-}
     Just f <*> Just x = Just (f x)
```

Comment:
We do not care about laziness at all and that is why this definition is strict, right?

Reply:
How would it be less strict? By putting the `Nothing` cases first?

Reply:
By putting the `Nothing <*> _ = Nothing` case first as it forces the first argument, but not the second one.

Reply:
Doesn't cost us anything, so sure.